x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/ReservedFieldNames.java

```diff
@@ -121,6 +121,7 @@
             CategoryDefinition.EXAMPLES.getPreferredName(),
             CategoryDefinition.NUM_MATCHES.getPreferredName(),
             CategoryDefinition.PREFERRED_TO_CATEGORIES.getPreferredName(),
+            CategoryDefinition.MLCATEGORY.getPreferredName(),
```

Comment:
I think this will stop the field getting written into anomaly records, which would be very bad.  It will still get written as `"blah_field_name": "mlcategory"`, `"blah_field_value": "42"`, but won't get written as `"mlcategory": "42"`.  So this line needs to be removed.

Reply:
🤔 When I ran this manually, it worked just fine. We don't validate the reserved field names ahead of time anywhere?

Reply:
Ah, I see, we don't fail the job or anything. The field is just not written.